Makefile

```diff
@@ -0,0 +1,46 @@
+.PHONY: uv-download
+uv-download:
+ifeq ($(OS),Windows_NT)
+	# Windows-specific uv installation
+	powershell -ExecutionPolicy ByPass -c "irm https://astral.sh/uv/install.ps1 | iex"
+else
+	# Unix-like systems (Linux/macOS)
+	curl -LsSf https://astral.sh/uv/install.sh | sh
+endif
+
+.PHONY: install-tests
+install-tests:
+	uv venv
+ifeq ($(OS),Windows_NT)
+	# Windows-specific virtual environment activation
+	. .venv/Scripts/activate && uv pip install -e .[tests]
```

Comment:
I think we should use "uv sync  --dev", rename "tests" by "dev" in pyproject.toml and move it to the new "[dependency-groups]" instead of "[project.optional-dependencies]"

Reply:
I don't think we are ready to move to [dependency-groups]:

https://github.com/astral-sh/uv/issues/8590#issuecomment-2439611796

The only one supporting it is https://hatch.pypa.io/latest/
but I wouldn't go that far.

Reply:
Updated the toml, see if good enough now